### PR TITLE
Use context manager for handling sessions

### PR DESCRIPTION
**Pull request by fridex**
Fixes: #1295

### CODE REVIEW

**Comment by fridex** [comment]
Testing... @pacospace meanwhile, PTAL

File: thoth/storages/graph/postgres.py [hunk 1]
```diff
@@ -3817,33 +3852,35 @@ def _create_python_software_stack(
             python_package_requirements = self._create_python_package_requirement(requirements)
```
Comment (pacospace) [inline thoth/storages/graph/postgres.py]: Missing `session`

Comment (fridex) [reply thoth/storages/graph/postgres.py]: Fixed! :+1:

File: thoth/storages/graph/postgres.py [hunk 2]
```diff
@@ -4123,31 +4150,33 @@ def _system_symbols_analysis_result(
...
                     )
 
                 DetectedSymbol.get_or_create(
-                    self._session,
+                    session,
                     package_extract_run_id=package_extract_run.id,
                     versioned_symbol_id=versioned_symbol.id
                 )
 
+    @staticmethod
```
Comment (pacospace) [inline thoth/storages/graph/postgres.py]: `_python_sync_analysis_result` is not a staticmethod :)

Comment (fridex) [reply thoth/storages/graph/postgres.py]: Nice! Fixed in 04d383f

**Pull request merged**